bundles/org.openhab.core.io.http.auth/src/main/java/org/openhab/core/io/http/auth/internal/AuthorizePageServlet.java

```diff
@@ -191,17 +199,25 @@ protected void doPost(@Nullable HttpServletRequest req, @Nullable HttpServletRes
                     user = userRegistry.register(username, password, Set.of(Role.ADMIN));
                     logger.info("First user account created: {}", username);
                 } else {
+                    // Enforce a dynamic cooldown period after a failed authentication attempt: the number of
+                    // consecutive failures in seconds
+                    if (lastAuthenticationFailure != null && lastAuthenticationFailure
+                            .isAfter(Instant.now().minus(Duration.ofSeconds(authenticationFailureCount)))) {
+                        throw new AuthenticationException("Too many consecutive login attempts");
```

Comment:
Will you evaluate this somehow in the UI? How will the user know that he is merely temporarily blocked for a few seconds? Maybe we could simply add that info ("Login possible again after x seconds" or something like that).

Reply:
It only says "Please try again" to the user but the real reason is logged as a warning so you would see "Authentication failed: Too many consecutive login attempts" in the logs. Isn't is enough?

Reply:
A "Please try again" does not tell the user much. But as I assume that a normal user won't misspell his password more than 3 times in a row, he won't really hit the delay at all - and for brute force attackers, I don't care what message they see ;-)

Reply:
Yes, in some schools of thought it's considered a good practice to remain rather opaque on what the actual issue is when the login fails - i.e. not tell that the user account actually exists but the password is wrong, or if the account is locked for some reason. It doesn't scale well when you have thousands of accounts but I believe we're not in that situation with openHAB so it's fine to look at the logs to figure out why a otherwise legitimate login didn't succeed ;)